server/src/main/java/org/candlepin/resource/HypervisorResource.java

```diff
@@ -341,8 +294,17 @@ public AsyncJobStatusDTO hypervisorHeartbeatUpdate(
             .setOwner(owner)
             .setReporterId(reporterId);
 
-        AsyncJobStatus job = this.jobManager.queueJob(config);
-        return this.translator.translate(job, AsyncJobStatusDTO.class);
+        try {
+            AsyncJobStatus job = this.jobManager.queueJob(config);
+            return this.translator.translate(job, AsyncJobStatusDTO.class);
+
+        }
+        catch (JobException e) {
+            String errmsg = this.i18n.tr("An unexpected exception occurred while scheduling job \"{0}\"",
+                config.getJobKey());
+
+            throw new IseException(errmsg, e);
```

Comment:
Same here: add  `log.error(errmsg, e)` before throwing

also, it seems you need to rebase the PR due to the RoleResource one being merged earlier

Reply:
Done.